### PR TITLE
Fixes https://github.com/lwansbrough/react-native-camera/issues/743

### PR DESCRIPTION
**Pull request by iegik**

### CODE REVIEW

**Comment by ptomasroos** [comment]
Thanks for this PR but how can the component be mounted ever in componentWillMount since the lifecycle explicitly in react is ctor -> cwm -> cdm ?

**Comment by iegik** [comment]
@ptomasroos In react lifecycle will be like it was described only if all code will by synchronized and consistent. In our case we have `async componentWillMount`, it means that `componentWillMount` will continue running while next step starts (cdm).

File: index.js [hunk 1]
```diff
@@ -160,10 +160,18 @@ export default class Camera extends Component {
 
     if (check) {
       const isAuthorized = await check();
-      this.setState({ isAuthorized });
+      if (this._isMounted) {
+        this.setState({ isAuthorized });
+      } else {
+        this.state.isAuthorized = isAuthorized;
```
Comment (chirag04) [inline index.js]: !

Comment (iegik) [reply index.js]: What do You propose? Delete this line? Call `setState` within `setTimeout`?
How to wait `render` is finished in RN, `requestAnimationFrame`?

**Pull request closed without merging**